azext_iot/operations/hub.py

```diff
@@ -1940,6 +1949,51 @@ def iot_get_sas_token(
     }
 
 
+def _iot_build_sas_token_from_cs(connection_string, duration=3600):
+    from azext_iot.common._azure import (
+        parse_iot_device_connection_string,
+        parse_iot_device_module_connection_string,
+        parse_pnp_connection_string,
+        parse_iot_hub_connection_string
+    )
+    uri = None
+    policy = None
+    key = None
+
+    parsed_cs = None
+    all_parsers = [
+        ("Module", parse_iot_device_module_connection_string),
+        ("Device", parse_iot_device_connection_string),
+        ("PnP Model Repository", parse_pnp_connection_string),
+        ("IoT Hub", parse_iot_hub_connection_string),
```

Comment:
Suggestion - we can make these constants (either in a shared file or even just locally) to avoid string matching errors in the logic below.